R/manifest.R

```diff
@@ -518,18 +571,22 @@ update_manifest_column <- function(dataflow_manifest,
                                    recalc_num_items = FALSE,
                                    access_token,
                                    base_url) {
-  # arrange by entityId
-  dataflow_manifest <- dplyr::arrange(dataflow_manifest, dataset_id)
-  get_all_manifests_out <- dplyr::arrange(get_all_manifests_out, dataset_id)
-
-  # get logical index of which items have changed
-  idx <- dataflow_manifest[, update_column] != get_all_manifests_out[, update_column]
+  # full join two datasets and filter on mismatched updated_column
```

Comment:
I was running into an edge case where the two input datasets had mismatched number of columns, resulting in a vector recycling warning. The cause was resolved by fixing a merge outside of this function, but it made me realize it's not safe to assume the input dataset records match.

By matching the datasets with a merge, it should be easier to guarantee that the updated columns are matched to the correct dataset_id